### ISSUE TITLE
Arithmetic in array subscript crashes T2

### ISSUE THREAD

**Issue by hhezart** [J2Kbr/GtunerIV#5]
The following statements will crash Titan 2:

<pre>
const fix32 arr[] = {1.0, 2.0, 3.0, 4.0};
int ii = 1;
fix32 p = arr[ii*2]; // crash
</pre>

**Comment by J2Kbr** [comment]
confirmed. To be fixed for the next Gtuner IV beta release.

**Comment by J2Kbr** [comment]
**More info from bug report 4:**

```
const int8 somearray[] = {0, 0, 5, 40, 20, 50, -20, 40, 0, 45, 5, 65};
int i = 2 * 2;
result = somearray[i]; // keeps returning 0, wrong value
result = somearray[4]; // numeric literal 4 returns the right value!
```

Update: removing the const fixed the problem.

non-const does a memcpy and subscript works

```
    MCPY    23E3,<000D>,000C
    ....
    LCW!    23E3
```

vs const where data seg is loaded but subscript doesn't work 
`LCW!    <000D>`

**Comment by J2Kbr** [comment]
Fixed on Gtuner IV 0.91 Beta
